hibernate-core/src/main/java/org/hibernate/dialect/temptable/SpannerTemporaryTableExporter.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.dialect.temptable;
+
+import org.hibernate.dialect.SpannerDialect;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * Spanner-specific exporter for temporary tables.
+ * Spanner requires PRIMARY KEY definitions to be outside the column definition list.
+ * * Syntax: CREATE TABLE Name (Col Type) PRIMARY KEY (Col)
```

Comment:
This is no longer a requirement. Spanner now supports inlined primary key constraints like this:

```sql
create table my_table (id int64 primary key, value string(max));
```